mcs/class/corlib/Test/System.IO/FileTest.cs

```diff
@@ -2680,5 +2682,59 @@ public void MoveTest ()
 			MoveTest (FileAccess.ReadWrite, FileShare.Write | FileShare.Delete, true);
 			MoveTest (FileAccess.ReadWrite, FileShare.ReadWrite | FileShare.Delete, true);
 		}
+
+
+		[DllImport ("libc", SetLastError=true)]
+		public static extern int symlink (string oldpath, string newpath);
+
+		[DllImport ("libc", SetLastError=true)]
+		public static extern int rename (string oldpath, string newpath);
+
+		static void DeleteBadSymLink (string path) {
+			string newpath = path + new Random ().Next ();
+			rename (path, newpath);
+			File.Delete (newpath);
+		}
+
+		[Test]
+		public void SymLinkLoop ()
+		{
+			if (!RunningOnUnix)
+				Assert.Ignore ("Symlink are hard on windows");
+
+			var name1 = "f0ew4fuweo";
+			var name2 = "jneruyhbv32ir";
```

Comment:
appending a random number would be good here too, we do run multiple builds and test runs on the same machine.